x-pack/qa/sql/src/main/java/org/elasticsearch/xpack/qa/sql/rest/RestSqlTestCase.java

```diff
@@ -319,7 +320,10 @@ private void expectBadRequest(CheckedSupplier<Map<String, Object>, Exception> co
             request.addParameter("mode", mode);        // JDBC or PLAIN mode
         }
         if (randomBoolean()) {
-            request.addHeader("Accept", randomFrom("*/*", "application/json"));
+            // JSON is the default but randomly set it sometime for extra coverage
+            RequestOptions.Builder options = request.getOptions().toBuilder();
```

Comment:
out of curiosity: here and in other tests why do we need to get the options and then modify them? Aren't those the default options? I would rather build new options and set them.

Reply:
They are the default options. I could do `RequestOptions.DEFAULTS.toBuilder()` just as well but this way feels a little more natural to me.

Reply:
I see. what happens if users forget to provide e.g. the consumer factory? I guess that is what would happen by building new options?

Reply:
I don't 100% understand your question but I think the answer is yes. The `request.getOptions().toBuilder()` means that if they already have something in the options then they are just modifying them.

I think it'd be worth me adding some more to the docs about these options. An example of making one that you share and something to make it more obvious that you should set these options in one go rather than making two builders. I mean, you can make two builder if you want, but it'll involve a couple of extra copy operations.

Reply:
my question was not clear. I thought there was a way to start with a new builder without providing e.g. the consumer factory and set some header. In that case I was wondering what the client would do. What I was getting to was "would it be possible to do without this get default, modify and set, rather just build new ones and set". we could have the default consumer factory as a default in the builder in case it's not set? and the headers would be empty by default anyway? Maybe I am missing something though :)

Reply:
I did it this way because "build new ones" amounts to the same thing as "create a builder from the DEFAULT instance". We want people to think of this as share-able so I made the API for building new ones always start with a shared "default" one.